dnf/transaction_sr.py

```diff
@@ -536,13 +536,18 @@ def run(self):
 
         for pkg_data in self._rpms:
             try:
+                action = pkg_data["action"]
+                if action is None:
+                    continue
```

Comment:
This is also not correct, we would miss logging the unexpected state transitions.